### PR TITLE
[SG-34733] Accessibility: Search box ARC checklist issues

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
### Audit type
ARC Toolkit

### User journey audit issue
[#34188](https://github.com/sourcegraph/sourcegraph/issues/34188)

### Problem description
- [x]  SVG icons inside of toggles should be aria-hidden
- [x] Search box Monaco editor should have role="textbox" (currently has role="code" which is invalid)

### Expected behavior
Issues fixed and no longer flagged by ARC toolkit

## Refs
[Sourcegraph Issue](https://github.com/sourcegraph/sourcegraph/issues/34733)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-34733)

## Test plan

- Install the ARC toolkit extension on the Chrome browser, Go to Developer Tools, select the ARC tab and run tests to retrieve the stats

## App preview:

- [Web](https://sg-web-contractors-sg-34733.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-vxckybigul.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/search-ui/src/input/MonacoQueryInput.tsx [hunk 1]
```diff
@@ -181,10 +181,14 @@ export const MonacoQueryInput: React.FunctionComponent<React.PropsWithChildren<M
 
     const onEditorCreated = useCallback(
         (editor: Monaco.editor.IStandaloneCodeEditor) => {
+            // `aria-label` and `role` set to fix accessibility issues
+            // https://github.com/sourcegraph/sourcegraph/issues/34733
+            editor.getDomNode()?.setAttribute('role', 'textbox')
+            editor.getDomNode()?.setAttribute('aria-label', placeholder || 'Enter your query...')
```
Comment (limitedmage) [inline client/search-ui/src/input/MonacoQueryInput.tsx:187]: @fkling Is something like this possible/needed for CodeMirror?

**Review by limitedmage — APPROVED** [review]

File: client/search-ui/src/input/MonacoQueryInput.tsx (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/search-ui/src/input/MonacoQueryInput.tsx]: Thanks @limitedmage we add new `ariaLabel` prop with `Search query`. Could we merge this now and we will take care `CodeMonitor` form in other issues

Comment (fkling) [reply client/search-ui/src/input/MonacoQueryInput.tsx]: @limitedmage CodeMirror already sets `role="textbox"` but not `aria-label`. That's the one I have to file an issue for CodeMirror.

Comment (fkling) [reply client/search-ui/src/input/MonacoQueryInput.tsx]: @limitedmage Looks like there is already a way to configure this, but that wasn't clear to me: https://github.com/codemirror/codemirror.next/issues/848#issuecomment-1143229718 . I'll take care of that as part of migrating the tests.

Comment (gitstart-sourcegraph) [reply client/search-ui/src/input/MonacoQueryInput.tsx]: @fkling @limitedmage Updated PR to make the same attributes on CodemirrorQueryInput component. Could you take a look?

**Review by limitedmage — APPROVED** [review]

**Pull request merged**